catalog/db_scanner.go

```diff
@@ -15,12 +15,11 @@ type DBScanner interface {
 	Err() error
 }
 
-func ScanDBEntryUntil(s DBScanner, ent *DBScannerEntry, p string) (*DBScannerEntry, error) {
-	for ent == nil || ent.Path < p {
+func ScanDBEntriesUntil(s DBScanner, p string) (*DBScannerEntry, error) {
+	for s.Value() == nil || s.Value().Path < p {
 		if !s.Next() {
 			return nil, s.Err()
 		}
-		ent = s.Value()
 	}
-	return ent, nil
+	return s.Value(), nil
```

Comment:
1. Can reduce the checks done calling Value multiple times by capture the value
2. No need to return the entry we just moved the scanner - caller can call Value()
```
ent := s.Value()
for ent == nil || ent.Path < p {
  if !s.Next() {
    return s.Err()
  }
  ent = s.Value()
}
return nil
```